src/unixTerminal.ts

```diff
@@ -73,10 +75,12 @@ export class UnixTerminal extends Terminal {
     };
 
     // fork
-    const term = pty.fork(file, args, parsedEnv, cwd, cols, rows, uid, gid, onexit);
+    const term = pty.fork(file, args, parsedEnv, cwd, cols, rows, uid, gid, (encoding === 'utf8'), onexit);
 
     this.socket = new PipeSocket(term.fd);
-    this.socket.setEncoding('utf8');
+    if (encoding !== null) {
+      this.socket.setEncoding(encoding);
```

Comment:
My impression is that this encoding setting isn't useful on Windows.  IIRC, on Unix, a pty has no inherent encoding (it's just bytes), and it looks like node-pty currently decodes the pty bytes to a JavaScript string.  On Windows, I think the console screen buffer is Unicode text underneath.  winpty reads UTF-16 text using `ReadConsoleOutputW` and converts it to UTF-8 for node-pty.